internal/gateway/connector/helpers.go

```
@@ -213,23 +249,74 @@ func packagedWindowsHookBinary(executable string) string {
 func packagedWindowsHookBinaryAtRoot(executable, expectedRoot string) string {
+	return packagedWindowsHookBinaryAtLayout(executable, expectedRoot, expectedRoot, false)
+}
+
+// packagedWindowsHookBinaryAtUninstallRoot recognizes the exact random trash
+// tree created by native setup after its durable uninstall commit. Connector
+// teardown runs the gateway from that tree because InstallRoot has already
+// been atomically removed. The embedded install state continues to bind every
+// logical path to expectedRoot, and only a strict `.uninstall.<128-bit hex>`
+// sibling is accepted. This lets teardown recognize the stable hook command
+// without trusting an environment variable or arbitrary relocated checkout.
+func packagedWindowsHookBinaryAtUninstallRoot(executable, expectedRoot string) string {
```

golangci-lint (2.12.2)

[error] func packagedWindowsHookBinaryAtUninstallRoot is unused

(unused)